R/produce_map.R

```diff
@@ -0,0 +1,64 @@
+#' Create an interactive html based map of animal locations.
+#'
+#' @param x A dataframe containing collar data with the columns "id", "lat", "lon" and  "date_time".
+#' @param save_html Logical, whether or not the map should be saved to disk.
+#' @param file Used only if save_html = T. The direcotry and file name of the output html map file.
+#'
+#' @return A leaflet map.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   collardata %>%
+#'   filter_last_loc() %>%
+#'   produce_map()
+#' }
+produce_map <- function(x,
+                        save_html = F,
+                        file = "map.html"
+) {
+  assertthat::assert_that("lat" %in%  names(x), msg = "A latitude column named 'lat' is required")
+  assertthat::assert_that("lon" %in%  names(x), msg = "A longitude column named 'lon' is required")
+  assertthat::assert_that("id" %in%  names(x), msg = "An id column named 'id' is required")
+  assertthat::assert_that("dt" %in%  names(x), msg = "A date time column named 'dt' is required")
+  assertthat::assert_that(assertthat::noNA(x$lon))
+  assertthat::assert_that(assertthat::noNA(x$lat))
+
+  if(interactive())
```

Comment:
I think the style guide would suggest using {} with all control statements. I would appreciate them as a reader.

Reply:
And thank you for adding the interactive call...